src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs

```diff
@@ -748,10 +748,16 @@ internal struct GCConfigurationContext
         }
 
         [UnmanagedCallersOnly]
-        private static unsafe void Callback(void* configurationContext, void* name, GCConfigurationType type, long data)
+        private static unsafe void Callback(void* configurationContext, void* name, void* publicKey, GCConfigurationType type, long data)
         {
+            // If the public key is null, it means that the corresponding configuration isn't publically available
+            // and therefore, we shouldn't add it to the configuration dictionary to return to the user.
+            if (publicKey == null)
+            {
+                return;
+            }
+
             Debug.Assert(configurationContext != null);
-            Debug.Assert(name != null);
```

Comment:
Intentional? The `Marshal.PtrToStringUTF8()` API does handle `null`, but returns `null` if that is what is passed in. However, usage of `!` indicates there is an assumption that `null` will never be returned which does seem to be true. The `Assert` doesn't prevent that but is good at signaling the code isn't assuming `null` is valid.

Reply:
Thanks! Definitely makes sense to add this back in to be safe.